gateway/observe.go

```diff
@@ -0,0 +1,15 @@
+package gateway
+
+import (
+	"github.com/buildwithgrove/path/observation"
+)
+
+// RequestResponseReporter defines the interface for reporting all the details
+// regarding a request and its corresponding response and set of events to any interested entity.
+// Examples of reporters includes:
+// - The MetricsReporter component: to export metrics based on the observations.
+// - The DataReporter: to export the observations to an external component: eg. a Messaging system or a Database.
```

Comment:
Always keep the reader in mind.

All I did was change how much reading I have to do (cause I also have a short attention span like everyone else). Now the next 1000 people are going to need to read less.


```suggestion
// RequestResponseReporter defines the interface for reporting observations with respect to
// a request, its corresponding response, and the set of events to any interested entity.
// Examples of reporters include:
// 	- MetricsReporter: exports metrics based on the observations
// 	- DataReporter: exports observations to external components (e.g.Messaging system or Database)
```